rts/motoko-rts/src/gc/incremental/phases/mark_increment.rs

```diff
@@ -0,0 +1,134 @@
+use crate::{
+    gc::incremental::{
+        array_slicing::slice_array,
+        mark_stack::MarkStack,
+        partitioned_heap::PartitionedHeap,
+        roots::{visit_roots, Roots},
+        time::BoundedTime,
+        State,
+    },
+    memory::Memory,
+    types::*,
+    visitor::visit_pointer_fields,
+};
+
+pub struct MarkState {
+    mark_stack: MarkStack,
+    complete: bool,
+}
+
+pub struct MarkIncrement<'a, M: Memory> {
+    mem: &'a mut M,
+    time: &'a mut BoundedTime,
+    heap: &'a mut PartitionedHeap,
+    mark_stack: &'a mut MarkStack,
+    complete: &'a mut bool,
```

Comment:
```suggestion
    mark_state: &'a mut MarkState
```
Would this be possible for clarity, i.e. just alias the MarkState, not its fields? Or does it make it slower?

Reply:
I introduced it primarily for convenience for the accesses inside the mark increment phase (to skip the navigation via state). I believe it should not be slower but will measure it.

Reply:
I measured it and the difference is negligibly small. Interestingly, the combined state with the extra indirection is sometimes a little bit faster (<0.1%).